src/main/java/com/bootcamp/demo/service/ScooterService.java

```diff
@@ -0,0 +1,157 @@
+import com.bootcamp.demo.model.component.Location;
+import com.bootcamp.demo.model.component.ScooterStatus;
+import com.bootcamp.demo.service.exception.ItemNotFoundException;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+@Service
+public class ScooterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScooterService.class);
+    private static final String COLLECTION_SCOOTERS_PATH = "bookings/databases/scooters";
+    private final Firestore db;
+
+    private ScooterService(Firestore db) {
+        this.db = db;
+    }
+
+
+    public Scooter findScooterById(String scooterId) {
+        try {
+            return db.collection(COLLECTION_SCOOTERS_PATH)
+                    .document(scooterId)
+                    .get()
+                    .get().toObject(Scooter.class);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error("Failed to find scooter with id = {}.", scooterId, e);
+            throw new ServiceException(e);
+        }
+    }
+
+
+    public Set<Scooter> findAllScooters() {
+        try {
+            return db.collection(COLLECTION_SCOOTERS_PATH)
+                    .get()
+                    .get()
+                    .getDocuments()
+                    .stream()
+                    .map(queryDocumentSnapshot -> queryDocumentSnapshot.toObject(Scooter.class))
+                    .collect(Collectors.toSet());
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error("Failed to find all scooters.", e);
+            throw new ServiceException(e);
+        }
+
+    }
+
+
+    public void createScooter(final Scooter scooter) {
+        try {
+            ApiFuture<WriteResult> collectionsApiFuture = db.collection(COLLECTION_SCOOTERS_PATH)
+                    .document(scooter.getSerialNumber())
+                    .set(scooter);
+
+            String updateTime = collectionsApiFuture.get().getUpdateTime().toDate().toString();
+
+            LOGGER.info("Created scooter with id = {} at = {}", scooter, updateTime);
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.error("Failed to create scooter with id = {}.", scooter, e);
+            throw new ServiceException(e);
+        }
+    }
+
+
+    public String updateScooter(final String scooterId, Location location, ScooterStatus newStatus, Double newBatteryLevel) throws ExecutionException, InterruptedException {
```

Comment:
Based on the comment from the BookingController, we could check if this is need or not. If kept, you could chain the calls and result in sth like, as it does not look like the variables are needed:
```
Scooter newScooter = new Scooter(scooterId, location, newBatteryLevel, newStatus);
return db.collection(COLLECTION_SCOOTERS_PATH)
        .document(scooterId)
        .set(newScooter)
        .get()
        .getUpdateTime()
        .toString();
```